opensquirrel/export/quantify_scheduler_exporter.py

```diff
@@ -0,0 +1,93 @@
+import math
+
+from opensquirrel.common import ATOL
+from opensquirrel.default_gates import x, z
+from opensquirrel.squirrel_ir import (
+    BlochSphereRotation,
+    ControlledGate,
+    MatrixGate,
+    Qubit,
+    SquirrelIR,
+    SquirrelIRVisitor,
+)
+
+try:
+    import quantify_scheduler
+    import quantify_scheduler.operations.gate_library as quantify_scheduler_gates
+except Exception as e:
+    pass
+
+
+class _ScheduleCreator(SquirrelIRVisitor):
+    def _get_qubit_string(self, q: Qubit) -> str:
+        return f"{self.qubit_register_name}[{q.index}]"
+
+    def __init__(self, qubit_register_name: str):
+        self.qubit_register_name = qubit_register_name
+        self.schedule = quantify_scheduler.Schedule(f"Exported OpenSquirrel circuit")
+
+    def visit_bloch_sphere_rotation(self, g: BlochSphereRotation):
+        if abs(g.axis[2]) < ATOL:
+            # Rxy rotation.
+            theta: float = g.angle
+            phi: float = math.atan2(g.axis[1], g.axis[0])
+            self.schedule.add(quantify_scheduler_gates.Rxy(theta=theta, phi=phi, qubit=self._get_qubit_string(g.qubit)))
+            return
+
+        if abs(g.axis[0]) < ATOL and abs(g.axis[1]) < ATOL:
+            # Rz rotation.
+            self.schedule.add(quantify_scheduler_gates.Rz(theta=g.angle, qubit=self._get_qubit_string(g.qubit)))
+            return
+
+        raise Exception(
+            "Cannot export circuit: it contains unsupported gates - decompose them to the "
+            "Quantify-scheduler gate set first"
+        )
+
+    def visit_matrix_gate(self, g: MatrixGate):
+        raise Exception("Unimplemented: would require a complicated mathematical decomposition...")
```

Comment:
If the IR would contain a matrix gate, this should then be decomposed first, but this is quite complex. So for people who want to export to a Quantify Schedule, they cannot define matrix gates?

Reply:
Actually the message is wrong.
In this design, the user is responsible for decomposing herself the circuit to CZ/RXY/RZ
So this message should just be "Quantify-scheduler exporter only supports CZ/RXY/RZ"
Concerning the MatrixGate, it is still possible, but since the CNOT decomposer doesn't decompose them (currently?), then decompositions have to be done on a case-by-case basis using `circuit.replace`, so hand-written decompositions for e.g. SWAP